pygfx/utils/transform.py

```diff
@@ -590,30 +590,38 @@ def __init__(
             self._reference_up = reference_up
         else:
             # use given reference_up (and sync own)
-            self._reference_up = reference_up
+            self._reference_up = np.asarray(reference_up, dtype=float)
             own_ref = la.vec_transform(reference_up, self.parent.inverse_matrix)
             self.own._reference_up = own_ref
 
-        self.parent.on_update(self.parent_updated)
-        self.own.on_update(self.child_updated)
+        self.parent.on_update(self._parent_updated)
+        self.own.on_update(self._own_updated)
 
     def flag_update(self):
         self.last_modified = perf_counter_ns()
         super().flag_update()
 
     @callback
-    def parent_updated(self, parent: AffineBase):
-        own_ref = la.vec_transform(self.reference_up, parent.inverse_matrix)
-        parent_origin = la.vec_transform((0, 0, 0), parent.inverse_matrix)
-        self.own._reference_up = own_ref - parent_origin
-
+    def _parent_updated(self, parent: AffineBase):
+        self._propagate_reference_up()
         self.flag_update()
 
+    def _propagate_reference_up(self):
+        new_ref = la.vec_transform(self.reference_up, self.parent.inverse_matrix)
+        origin = la.vec_transform((0, 0, 0), self.parent.inverse_matrix)
+        self.own._reference_up = new_ref - origin
```

Comment:
does this resulting vector potentially need normalizing?

Reply:
Good point.

Reply:
Done!